### PR TITLE
303 Add programme details confirmation component

### PR DESCRIPTION
**Pull request by SamJamCul**
### Context
https://trello.com/c/9r0pBHJR/303-s-confirm-trainee-programme-details-component
### Changes proposed in this pull request
* Added a component for the programme details confirmation
* Added a test and preview for this component
### Guidance to review

Component preview 

image

### CODE REVIEW

**Comment by SamJamCul** [comment]
[[S] Confirm Trainee Programme Details Component](https://trello.com/c/9r0pBHJR/303-s-confirm-trainee-programme-details-component)

File: app/components/trainees/confirmation/programme_details/view.html.erb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      {
+        key: "Subject",
+        value: subject,
+        action: govuk_link_to('Change<span class="govuk-visually-hidden"> subject</span>'.html_safe, 
+                              edit_trainee_programme_details_path(trainee))
+      },
+      {
+        key: "Age Range",
+        value: age_range,
+        action: govuk_link_to('Change<span class="govuk-visually-hidden"> age_range</span>'.html_safe,
```
Comment (aliuk2012) [inline app/components/trainees/confirmation/programme_details/view.html.erb:13]:
```suggestion
        action: govuk_link_to('Change<span class="govuk-visually-hidden"> age range</span>'.html_safe,
```

File: app/components/trainees/confirmation/programme_details/view.html.erb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+                              edit_trainee_programme_details_path(trainee))
+      },
+      {
+        key: "Programme Start Date",
+        value: programme_start_date,
+        action: govuk_link_to('Change<span class="govuk-visually-hidden"> progamme_start_date</span>'.html_safe,
```
Comment (aliuk2012) [inline app/components/trainees/confirmation/programme_details/view.html.erb:19]:
```suggestion
        action: govuk_link_to('Change<span class="govuk-visually-hidden"> progamme start date</span>'.html_safe,
```

File: app/components/trainees/confirmation/programme_details/view.html.erb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    heading_level: 2,
+    rows: [
+      {
+        key: "Subject",
+        value: subject,
+        action: govuk_link_to('Change<span class="govuk-visually-hidden"> subject</span>'.html_safe, 
+                              edit_trainee_programme_details_path(trainee))
+      },
+      {
+        key: "Age Range",
```
Comment (aliuk2012) [inline app/components/trainees/confirmation/programme_details/view.html.erb:11]:
```suggestion
        key: "Age range",
```

File: app/components/trainees/confirmation/programme_details/view.html.erb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        value: age_range,
+        action: govuk_link_to('Change<span class="govuk-visually-hidden"> age_range</span>'.html_safe,
+                              edit_trainee_programme_details_path(trainee))
+      },
+      {
+        key: "Programme Start Date",
```
Comment (aliuk2012) [inline app/components/trainees/confirmation/programme_details/view.html.erb:17]:
```suggestion
        key: "Programme start date",
```

File: app/components/trainees/confirmation/programme_details/view.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
+module Trainees
+  module Confirmation
+    module ProgrammeDetails
+      class View < GovukComponent::Base
+        include SanitizeHelper
```
Comment (aliuk2012) [inline app/components/trainees/confirmation/programme_details/view.rb:5]: I think we can remove this because we don't seem to be using it

**Review by aliuk2012 — CHANGES_REQUESTED** [review]

File: app/components/trainees/confirmation/programme_details/view.rb (hunk 5 above)
Comment (SamJamCul) [reply app/components/trainees/confirmation/programme_details/view.rb:5]: Ah yeah, my bad, just copied it across!

File: spec/components/trainees/confirmation/programme_details/view_spec.rb [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+          it "renders blank rows for subject, age range, and programme start date" do
+            expect(component).to have_selector(".govuk-summary-list__row", count: 3)
+          end
+
+          it "tells the user that no data has been entered" do
+            component.find_all(".govuk-summary-list__row").each do |row|
+              expect(row.find(".govuk-summary-list__value")).to have_text(t("components.confirmation.not_provided"))
+            end
+          end
```
Comment (defong) [inline spec/components/trainees/confirmation/programme_details/view_spec.rb:24]:
```suggestion

          it "tells the user that no data has been entered for subject, age range, and programme start date" do
            found = component.find_all(".govuk-summary-list__row")
            
            expect(found).to have size 3
            
            found.each do |row|
              expect(row.find(".govuk-summary-list__value")).to have_text(t("components.confirmation.not_provided"))
            end
          end
```
``` ruby
it "renders blank rows for subject, age range, and programme start date" do
# seems a bit pointless
```

Comment (SamJamCul) [reply spec/components/trainees/confirmation/programme_details/view_spec.rb:24]: Done, although this little bit isn't as functional as you implied! `expect(found).to have size 3`

**Review by aliuk2012 — APPROVED** [review]

**Review by darokel — APPROVED** [review]

**Comment by SamJamCul** [comment]
[[S] Confirm Trainee Programme Details Component](https://trello.com/c/9r0pBHJR/303-s-confirm-trainee-programme-details-component)

**Pull request merged**